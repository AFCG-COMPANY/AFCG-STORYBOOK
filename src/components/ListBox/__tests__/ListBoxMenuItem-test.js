import React from 'react';
import { mount } from 'enzyme';
import ListBox from '../';

describe('ListBoxMenuItem', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      isActive: false,
      isHighlighted: false,
      children: <span>ListBox.MenuItem</span>,
    };
  });

  it('should render', () => {
    const wrapper = mount(<ListBox.MenuItem {...mockProps} />);
    const activeWrapper = mount(
      <ListBox.MenuItem {...mockProps} isActive={true} />
    );
    const highlightedWrapper = mount(
      <ListBox.MenuItem {...mockProps} isHighlighted={true} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(activeWrapper).toMatchSnapshot();
    expect(highlightedWrapper).toMatchSnapshot();
  });
});
