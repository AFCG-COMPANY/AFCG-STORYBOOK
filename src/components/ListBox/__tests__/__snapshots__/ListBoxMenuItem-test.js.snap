// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBoxMenuItem should render 1`] = `
<ListBoxMenuItem
  isActive={false}
  isHighlighted={false}
>
  <div
    className="bx--list-box__menu-item"
  >
    <span>
      ListBox.MenuItem
    </span>
  </div>
</ListBoxMenuItem>
`;

exports[`ListBoxMenuItem should render 2`] = `
<ListBoxMenuItem
  isActive={true}
  isHighlighted={false}
>
  <div
    className="bx--list-box__menu-item bx--list-box__menu-item--active"
  >
    <span>
      ListBox.MenuItem
    </span>
  </div>
</ListBoxMenuItem>
`;

exports[`ListBoxMenuItem should render 3`] = `
<ListBoxMenuItem
  isActive={false}
  isHighlighted={true}
>
  <div
    className="bx--list-box__menu-item bx--list-box__menu-item--highlighted"
  >
    <span>
      ListBox.MenuItem
    </span>
  </div>
</ListBoxMenuItem>
`;
